### ISSUE TITLE
GitVersion field in version 1.16.5 and 1.15.8 release binaries doesn't match patch version

### ISSUE THREAD

**Issue by lachie83** [kubernetes/release#1020]
Observed the following change in behavior on Kubernetes releases 1.16.5 and 1.15.8 specifically (I know the release team are working on tooling issues so just wanted to draw attention to this). The `GitVersion` field doesn't match the release version directory `kubectl` was pulled from. I currently assume this is happening for all binaries for the release but have only checked `kubectl`
```
$ curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.16.5/bin/darwin/amd64/kubectl && chmod +x ./kubectl && ./kubectl version --client
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 46.5M  100 46.5M    0     0  28.3M      0  0:00:01  0:00:01 --:--:-- 28.3M
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.6-beta.0", GitCommit:"e7f962ba86f4ce7033828210ca3556393c377bcc", GitTreeState:"clean", BuildDate:"2020-01-15T08:26:26Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"darwin/amd64"}
```

```
$ curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.15.8/bin/darwin/amd64/kubectl && chmod +x ./kubectl && ./kubectl version --client
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 46.3M  100 46.3M    0     0  24.1M      0  0:00:01  0:00:01 --:--:-- 24.1M
Client Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.9-beta.0", GitCommit:"14ede42c4fe699a7078b566d89abc160f26857a2", GitTreeState:"clean", BuildDate:"2020-01-15T11:15:40Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"darwin/amd64"}
```

What this means is that `https://storage.googleapis.com/kubernetes-release/release/v1.15.9-beta.0/bin/darwin/amd64/kubectl` and `https://storage.googleapis.com/kubernetes-release/release/v1.15.8/bin/darwin/amd64/kubectl` have the same `GitVersion`. Which represents a change in behavior

**Comment by justaugustus** [comment]
Live discussion happening here: https://kubernetes.slack.com/archives/CJH2GBF7Y/p1579199518125500?thread_ts=1579199518.125500&cid=CJH2GBF7Y

**Comment by dlisin** [comment]
seems hyperkube docker image is also affected:
```
>>> docker run k8s.gcr.io/hyperkube-amd64:v1.15.8 /hyperkube kubelet --version                                                                                              
Kubernetes v1.15.9-beta.0
```
```
>>> docker run --rm k8s.gcr.io/hyperkube-amd64:v1.16.5 /hyperkube kubelet --version                                                                                              
Kubernetes v1.16.6-beta.0
```

**Comment by justaugustus** [comment]
All artifacts of our release would be affected.

**Comment by ialidzhikov** [comment]
I think wrong tags also breaks the comparison in helm chars.

For example the following 
```
{{- if semverCompare ">= 1.16" .Capabilities.KubeVersion.GitVersion }}
# ...
{{- end }}
```

works perfectly on `v1.16.4`, but it does not work on `v1.16.5`.  😕

**Comment by justaugustus** [comment]
Thanks. We're discussing a fix here: https://github.com/kubernetes/release/pull/1021

**Comment by sandeepmendiratta** [comment]
Do we know when the release 1.15.9 will be cut (available) for use?

**Comment by justaugustus** [comment]
We're _tentatively_ planning for today. We're in the process of doing some additional checks in our build tool.

**Comment by justaugustus** [comment]
The Patch Release Team will be releasing new versions of Kubernetes (1.17.2, 1.16.6, 1.15.9) next week to fix this.
We're tentatively planning for Tuesday, January 21st, by EOD US PT, but given this week's debugging of our build tools, it's possible that this date may shift.

As of today, the build versions that we're targeting are as follows:
- v1.17.2-beta.0.2+59603c6e503c87 (Built at https://github.com/kubernetes/kubernetes/commit/59603c6e503c87169aea6106f57b9f242f64df89)
- v1.16.6-beta.0.3+72c30166b2105c (Built at https://github.com/kubernetes/kubernetes/commit/72c30166b2105cd7d3350f2c28a219e6abcd79eb)
- v1.15.9-beta.0.3+2e808b7cb054ee (Built at https://github.com/kubernetes/kubernetes/commit/2e808b7cb054ee242b68e62455323aa783991f03)

Those commits represent essentially no-op PRs we used to trigger a new build version in CI:
- https://github.com/kubernetes/kubernetes/pull/87334
- https://github.com/kubernetes/kubernetes/pull/87333
- https://github.com/kubernetes/kubernetes/pull/87332

The new releases will be functionally equivalent to this week's releases, save the few commits we've added to fix the versioning issue:
- https://github.com/kubernetes/kubernetes/compare/v1.17.1...59603c6
- https://github.com/kubernetes/kubernetes/compare/v1.16.5...72c3016
- https://github.com/kubernetes/kubernetes/compare/v1.15.8...2e808b7

**_When 1.17.2, 1.16.6, and 1.15.9 are released, please IMMEDIATELY use them instead of 1.17.1, 1.16.5, and 1.15.8._**

Also sent to kubernetes-dev mailing list: https://groups.google.com/d/topic/kubernetes-dev/Mhpx-loSBns/discussion

**Comment by justaugustus** [comment]
Releases have been staged in advance of Tuesday.

The following release tracking issues have been opened:
- 1.17.2: https://github.com/kubernetes/sig-release/issues/964
- 1.16.6: https://github.com/kubernetes/sig-release/issues/958
- 1.15.9: https://github.com/kubernetes/sig-release/issues/957

Links to GCB runs have been attached to each of these issues.
(Note that viewing logs is restricted to Release Managers.)

**Comment by lachie83** [comment]
This has been fixed in 1.16.6 and 1.15.9.

```
$ curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.15.9/bin/darwin/amd64/kubectl && chmod +x ./kubectl && ./kubectl version --client
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 46.3M  100 46.3M    0     0  27.4M      0  0:00:01  0:00:01 --:--:-- 27.4M
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.9", GitCommit:"2e808b7cb054ee242b68e62455323aa783991f03", GitTreeState:"clean", BuildDate:"2020-01-18T23:33:14Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"darwin/amd64"}
```
```
$ curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.16.6/bin/darwin/amd64/kubectl && chmod +x ./kubectl && ./kubectl version --client
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 46.5M  100 46.5M    0     0  26.8M      0  0:00:01  0:00:01 --:--:-- 26.7M
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.6", GitCommit:"72c30166b2105cd7d3350f2c28a219e6abcd79eb", GitTreeState:"clean", BuildDate:"2020-01-18T23:31:31Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"darwin/amd64"}
```

**Comment by justaugustus** [comment]
Thanks for closing this out, @lachie83!
